runtime/src/bank.rs

```diff
@@ -1382,12 +1382,61 @@ impl Bank {
         let (_, fill_sysvar_cache_time_us) = measure_us!(new
             .transaction_processor
             .fill_missing_sysvar_cache_entries(&new));
-        time.stop();
 
+        let (num_accounts_modified_this_slot, populate_cache_for_accounts_lt_hash_us) = new
```

Comment:
what about the sysvar update happened above? 
I assume that no tx will write to them. so we are fine?

Reply:
so if I understand correctly, one reason for the bad hash calculation is due to writing the accounts before tx processing; And the other reason is that when we inspect the account and populate old accounts during transaction accounts loading, we are *NOT* check for "strict ancestor". That's how we end up populating a bad account in the cache. 

This PR fix the first issue and prevent us from adding those updated accounts in the cache. 

I wonder if we should introduce another check in inspect accounts - i.e. only populate the cache when the account is in from strict ancestors, just to be more defensive?

Reply:
> I wonder if we should introduce another check in inspect accounts - i.e. only populate the cache when the account is in from strict ancestors, just to be more defensive?

To be clear, this suggestion can be done in a follow up PR.